### PR TITLE
Update to use newer QPUs

### PR DESCRIPTION
**Pull request by JoelPasvolsky**
I mostly stuck to updating the examples with D-Wave 2000Q to Advantage and resisted restructuring and rewriting the surroundings wholesale, except a few places where I was overcome.

### CODE REVIEW

File: dwave/cloud/config.py [hunk 1]
```diff
@@ -48,171 +48,141 @@
...
+        [defaults]
+        token = ABC-123456789123456789123456789
 
-    ``DWAVE_API_TOKEN``: API authorization token.
+        [default-solver]
+        solver = {"qpu": true, "num_qubits__gt": 5000}
 
-    ``DWAVE_API_SOLVER``: Default solver.
+        [hybrid]
+        client = hybrid
```
Comment (randomir) [inline dwave/cloud/config.py:81]: Probably more useful config would be:
```suggestion
        [bqm]
        client = hybrid
        solver = {"supported_problem_types__contains": "bqm"}

        [cqm]
        client = hybrid
        solver = {"supported_problem_types__contains": "cqm"}
```
(`client` is optional above, but can leave it for explicitness)

Then:
```
>>> with Client.from_config(profile="cqm") as client:
...     cqm_solver = client.get_solver()
```

File: dwave/cloud/config.py [hunk 2]
```diff
@@ -48,171 +48,141 @@
...
     >>> import os
-    >>> os.environ['DWAVE_API_SOLVER'] = 'EXAMPLE_2000Q_SYSTEM'   # doctest: +SKIP
+    >>> os.environ["DWAVE_API_SOLVER"] = "Advantage_Emulator"   # doctest: +SKIP
     >>> dwave.cloud.config.load_config("./dwave_c.conf")   # doctest: +SKIP
-    {'client': 'sw',
-     'endpoint': 'https://url.of.some.dwavesystem.com/sapi',
-     'proxy': None,
-     'headers': None,
-     'solver': 'EXAMPLE_2000Q_SYSTEM',
+    {'solver': '{"software": true, "name": "Advantage_Emulator"}',
```
Comment (randomir) [inline dwave/cloud/config.py:182]: Haven't checked, but this should be:
```suggestion
    {'solver': 'Advantage_Emulator'},
```
If it's not, that might be a bug. Solver from the env should replace the complete solver definition, not just patch partially (name).

File: dwave/cloud/config.py [hunk 3]
```diff
@@ -48,171 +48,141 @@
...
+    https://eu-central-1.cloud.dwavesys.com/sapi/v2/
+    >>> print(client.token)  # doctest: +SKIP
+    DEF-987654321987654321987654321
     >>> # code that uses client
     >>> client.close() # doctest: +SKIP
 
-    The next example uses :func:`~dwave.cloud.config.load_config` to load profile values.
-    **Most users do not need to use this method.** It loads from the following configuration
-    file, dwave_c.conf, located in the current working directory, and specified explicitly::
+    *   **Advanced Example:** Loading profile values
```
Comment (randomir) [inline dwave/cloud/config.py:153]: I would just drop this complete section. Looks too complicated to be useful, as it's artificial and not describing a use case someone might have in practice (you would never override the envvar from code when you can just use kwargs to client init).

File: dwave/cloud/config.py [hunk 4]
```diff
@@ -48,171 +48,141 @@
...
-     'solver': 'EXAMPLE_2000Q_SYSTEM',
+    {'solver': '{"software": true, "name": "Advantage_Emulator"}',
      'token': 'ABC-123456789123456789123456789'}
-    >>> dc.config.load_config("./dwave_c.conf", profile='dw2000b', solver='Solver3')   # doctest: +SKIP
-    {'client': 'qpu',
-     'endpoint': 'https://url.of.some.dwavesystem.com/sapi',
-     'proxy': None,
-     'headers': None,
-     'solver': 'Solver3',
+    >>> dwave.cloud.config.load_config("./dwave_c.conf", profile="advantage-b", solver="Advantage_system4.1")   # doctest: +SKIP
```
Comment (randomir) [inline dwave/cloud/config.py:184]: I don't want to encourage any user to try something like this (use `load_config`). At least, not with the current client/config structure.

File: dwave/cloud/config.py [hunk 5]
```diff
@@ -490,29 +460,27 @@ def load_config_from_files(filenames=None):
     Examples:
         This example loads configurations from two files. One contains a default
         section with key/values that are overwritten by any profile section that
-        contains that key/value; for example, profile dw2000b in file dwave_b.conf
-        overwrites the default URL and client type, which profile dw2000a inherits
-        from the defaults section, while profile dw2000a overwrites the API token that
-        profile dw2000b inherits.
+        contains that key/value; for example, profile advantage_b in file dwave_b.conf
```
Comment (randomir) [inline dwave/cloud/config.py:463]: Can we please just drop this whole example? The `load_config_from_files` is such a low-level internal function we don't want anyone ever use it :smile: 

OTOH, we don't have to modify docs significantly here, as the whole config module will be replaced in #426.

File: dwave/cloud/config.py (hunk 3 above)
Comment (JoelPasvolsky) [reply dwave/cloud/config.py]: Dropped

File: dwave/cloud/config.py (hunk 2 above)
Comment (JoelPasvolsky) [reply dwave/cloud/config.py]: Dropped example

File: dwave/cloud/config.py (hunk 4 above)
Comment (JoelPasvolsky) [reply dwave/cloud/config.py]: Eliminated

**Pull request merged**